doc/read-the-docs-site/howtos/exporting-a-blueprint.rst

```diff
@@ -0,0 +1,403 @@
+.. highlight:: haskell
+.. _exporting_a_blueprint:
+
+How to produce a Plutus Contract Blueprint
+==========================================
+
+Plutus Contract Blueprints (`CIP-0057`_) are used to document the binary interface of a
+Plutus contract in a machine-readable format (JSON schema).
+
+A contract Blueprint can be produced by using the
+`writeBlueprint` function exported by the `PlutusTx.Blueprint` module::
+
+  writeBlueprint
+    :: FilePath
+    -- ^ The file path where the blueprint will be written to,
+    --  e.g. '/tmp/plutus.json'
+    -> ContractBlueprint
+    -- ^ Contains all the necessary information to generate 
+    -- a blueprint for a Plutus contract.
+    -> IO ()
+
+In order to demonstrate the usage of the `writeBlueprint` function,
+Let's consider the following example validator function and its interface:
+
+.. literalinclude:: Cip57Blueprint.hs
+   :start-after: BEGIN interface types
+   :end-before:  END interface types
+
+.. literalinclude:: Cip57Blueprint.hs
+   :start-after: BEGIN validator
+   :end-before:  END validator
+
+First of all we need to import required functionality:
+
+.. literalinclude:: Cip57Blueprint.hs
+   :start-after: BEGIN imports
+   :end-before:  END imports
+
+Next we define a contract blueprint value of the following type:
```

Comment:
sigh, this should all just be haddock links, but we still have no good haddock hosting solution :(

Reply:
Actually, we have this hosted https://intersectmbo.github.io/plutus/master/plutus-tx/html/PlutusTx-Blueprint-Contract.html#t:ContractBlueprint

Should I link it instead of embedding?